### PR TITLE
common/dns_resolve: fix memory leak

### PR DESCRIPTION
**Pull request by yaozongyou**
Signed-off-by: Yao Zongyou <yaozongyou@vip.qq.com>

### CODE REVIEW

File: src/common/dns_resolve.cc [hunk 1]
```diff
@@ -196,6 +196,11 @@ int DNSResolver::resolve_ip_addr(CephContext *cct, const string& hostname,
   if (r < 0) {
     return r;
   }
+
```
Comment (tchaikov) [inline src/common/dns_resolve.cc]: nit, might want to remove the empty lines.

Comment (yaozongyou) [reply src/common/dns_resolve.cc]: @tchaikov the empty line has been removed

File: src/common/dns_resolve.cc [hunk 2]
```diff
@@ -196,6 +196,10 @@ int DNSResolver::resolve_ip_addr(CephContext *cct, const string& hostname,
   if (r < 0) {
     return r;
   }
+  auto put_state = make_scope_guard([res, this] {
+      this->put_state(res);
+    });
+
```
Comment (tchaikov) [inline src/common/dns_resolve.cc]: could you remove this empty line also?

Comment (yaozongyou) [reply src/common/dns_resolve.cc]: removed

**Comment by shinobu-x** [comment]
test this please

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2017-12-23_05:58:55-rados-wip-kefu-testing-2017-12-23-1241-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2017-12-24_02:41:14-rados-wip-kefu-testing-2017-12-23-1241-distro-basic-mira/

the coredump is tracked by http://tracker.ceph.com/issues/22527

**Pull request merged**